ofrak_core/tests/components/test_android_sparse_component.py

```diff
@@ -0,0 +1,248 @@
+                proc = await asyncio.create_subprocess_exec(
+                    *cmd,
+                    stdout=asyncio.subprocess.PIPE,
+                    stderr=asyncio.subprocess.PIPE,
+                )
+                stdout, stderr = await proc.communicate()
+
+                if proc.returncode == 0:
+                    with open(raw_file.name, "rb") as f:
+                        raw_data = f.read()
+                    assert raw_data == self.EXPECTED_REPACKED_DATA
+                else:
+                    rc = proc.returncode or -1
+                    raise CalledProcessError(rc, cmd, stderr)
+
+    async def modify(self, unpacked_root_resource: Resource):
+        resource_to_modify = await unpacked_root_resource.get_only_child()
+        current_data = await resource_to_modify.get_data()
+        resource_to_modify.queue_patch(
+            Range.from_size(0, len(current_data)), self.EXPECTED_REPACKED_DATA
+        )
+        await resource_to_modify.save()
+
+
+@pytest.mark.skipif_missing_deps([AndroidSparseImagePacker, AndroidSparseImageUnpacker])
+class TestAndroidSparseImageUnpackModifyPack(AndroidSparseUnpackModifyPackPattern):
+    """
+    Test the basic unpack, modify, and pack functionality for Android sparse images.
+
+    This test requires simg2img tool to be installed and real test assets.
+    User must provide: tests/components/assets/hello_world.simg
+    """
+
+
+async def test_sparse_image_identification(ofrak_context: OFRAKContext):
+    """
+    Test that Android sparse images are correctly identified by magic bytes (REQ1.3).
+
+    This test verifies that:
+    - Files with sparse image magic bytes are identified as AndroidSparseImage
+    - The MagicIdentifier correctly tags the resource
+
+    User must provide: tests/components/assets/ext4.4096.simg
+    """
+    sparse_path = ASSETS_DIR / "ext4.4096.simg"
+
+    resource = await ofrak_context.create_root_resource_from_file(str(sparse_path))
+    await resource.identify()
+
+    assert resource.has_tag(AndroidSparseImage)
+
+
+@pytest.mark.skipif_missing_deps([AndroidSparseImagePacker, AndroidSparseImageUnpacker])
+async def test_large_sparse_image_roundtrip(ofrak_context: OFRAKContext):
+    """
+    Test unpacking and repacking of a larger sparse image (REQ4.4).
+
+    This test verifies that:
+    - Larger sparse images can be unpacked successfully
+    - The unpacked data maintains integrity
+    - The data can be repacked into a valid sparse image
+
+    User must provide: tests/components/assets/random8M.simg
+    """
+    large_sparse = ASSETS_DIR / "random8M.simg"
+
+    # Test unpacking
+    resource = await ofrak_context.create_root_resource_from_file(str(large_sparse))
+
+    await resource.unpack()
+    child = await resource.get_only_child()
+    unpacked_data = await child.get_data()
+
+    # Test repacking
+    await resource.pack()
+    repacked_data = await resource.get_data()
```

Comment:
Pedantic, but this is not testing, simply doing
```suggestion
    # Unpack
    resource = await ofrak_context.create_root_resource_from_file(str(large_sparse))
    await resource.unpack()
    child = await resource.get_only_child()
    unpacked_data = await child.get_data()

    # Repack
    await resource.pack()
    repacked_data = await resource.get_data()
```